crates/doctor/src/wasm.rs

```diff
@@ -11,34 +11,41 @@ use crate::{Diagnosis, Diagnostic, PatientApp};
 
 /// PatientWasm represents a Wasm source to be checked for problems.
 #[derive(Debug)]
-pub struct PatientWasm(RawComponentManifest);
+pub struct PatientWasm {
+    app_dir: PathBuf,
+    component: RawComponentManifest,
+}
 
 #[allow(missing_docs)] // WIP
 impl PatientWasm {
+    pub fn new(app_dir: impl AsRef<Path>, component: RawComponentManifest) -> Self {
```

Comment:
Making this `pub` makes it more difficult to extend `PatientWasm` in the future.

Reply:
Privatised